### ISSUE TITLE
Back-Button doesn't return to previous app

### ISSUE THREAD

**Issue by francwalter** [k9mail/k-9#963]
Hallo 
since long this issue in k9mail is present: when I open k9mail through a mail notification in the info bar, when having any other app opened, and after reading that mail, I go back with the Back-Key, then this won't bring me to the last opened app (from where I opened the notification), this brings me to the home screen. This is not the normal behaviour I think, other apps don't "forget" from where they have been started.
The same thing is, when I have an app open and I open now k9mail through some kind of gesture (there are several apps doing this, e.g. "Swapps"). I open k9 now and when I go back with the back button, the original app has gone, I come to the home screen.
The same issue I found with Threema, I guess it is the same bug.
But most other apps will bring me to the last screen (when I press the backbutton), from where they had been started, either through notification shortcut or other way I opened them, always I come back with the back button to the previous screen of any previous opened app.
Thank

frank

**Comment by cketti** [comment]
This behavior is intentional and recommended practice. See http://developer.android.com/training/notify-user/navigation.html and http://developer.android.com/design/patterns/navigation.html

**Comment by francwalter** [comment]
Yes, thank you for that second link. Here I read:
> The system Back button is used to navigate, in reverse chronological order, through the history of screens the user has recently worked with. It is generally based on the temporal relationships between screens, rather than the app's hierarchy.

And this is not done when I open k9 through a notification (when having another app actually open in foreground).
Do I misunderstand this feature?

**Comment by cketti** [comment]
Section "Navigation into Your App via Home Screen Widgets and Notifications":

> In the case of the Back button, you should make navigation more predictable by inserting into the task's back stack the complete upward navigation path to the app's topmost screen. This allows users who've forgotten how they entered your app to navigate to the app's topmost screen before exiting.

**Comment by francwalter** [comment]
> ... before exiting 

this means exiting *your app*, now it should return to the previous app.
That is how I understand this sentence.

**Comment by clawoflight** [comment]
@francwalter The meaning of that sentence is that the complete navigation path down to the current screen should be added to the stack when the user entered the app through another path, like a notification.
That way, even if they forgot that they had opened the app by clicking on e.g. a notification, they can still press the back button and reach the home screen of the app.

It says nothing about navigating back to the *previous app,* and I don't think that that is the desired behaviour for the back button.

**Comment by francwalter** [comment]
> It says nothing about navigating back to the previous app, and I don't think that that is the desired behaviour for the back button.
No, but it neither says, that the Back Button should go to Home after reaching the top of "Your App".

I just notice that every single app beside k9 (and Threema) does it like that (back button goes back to previous app) and now, that I know what is behind the k9-interpretation of user experience, I think it is a misunderstanding of the  recommended practice.

**Comment by francwalter** [comment]
Never mind. As I know this unusual behaviour in k9mail (and Threema!) I am used to it and when I think about opening k9mail through a notification in the info bar I never do this, when I dont want to leave the actual app in foreground.
The many many advantages in k9 are compensing vastly this little (in my eyes) bug in user experience abnormality.
Thank's!

**Comment by francwalter** [comment]
As this behaviour is also in Signal, I opened there[ an issue](https://github.com/signalapp/Signal-Android/issues/7362) too.
I still hope that this misinterpretation of Android Design Patterns are fixed in k9mail.

**Comment by francwalter** [comment]
After long time of annoyement about this "misinterpretation" I found [FairEmail](https://github.com/M66B/FairEmail), the E-Mail-Client from [Marcel Bokhorst](https://github.com/M66B). 
I know this developer from two great apps I use and used: NetGuard (one of his best works ever, I consider) and XPrivacy (dont use it anymore, it was with Xposed, which doesnt work on Android 9+).
FairEmail does indeed return to the last opened app, from where I tapped a notification about a new mail in the NotificationBar (like WhatsApp and Telegram too). 

So this interpretation of Android's Design Pattern is also a kind of interpretation of user experience.
I will change to FairEmail for test, looking if the rest of this email-client is as user friendly as this topic :)
By the way: FairEmail has also another feature which is annoying enough in k9mail, the possibility of size-reduce of attached images, see the very old [issue #3845](https://github.com/k9mail/k-9/pull/3845) which was never merged.
